ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerUtils.java

```diff
@@ -34,47 +37,39 @@
   private ServerUtils() {
   }
 
-  public static void handleError(final HttpServerResponse response, final int statusCode,
-      final int errorCode, final String errMsg) {
-    final ErrorResponse errorResponse = new ErrorResponse(errorCode, errMsg);
-    final Buffer buffer = errorResponse.toBuffer();
-    response.setStatusCode(statusCode).end(buffer);
-  }
-
-  public static void unhandledExceptionHandler(final Throwable t) {
-    log.error("Unhandled exception", t);
-  }
-
   public static <T> Optional<T> deserialiseObject(final Buffer buffer,
-      final HttpServerResponse response,
+      final RoutingContext routingContext,
       final Class<T> clazz) {
-    return PojoCodec.deserialiseObject(buffer, new HttpResponseErrorHandler(response), clazz);
+    return PojoCodec.deserialiseObject(buffer, new HttpResponseErrorHandler(routingContext), clazz);
   }
 
   private static class HttpResponseErrorHandler implements PojoDeserializerErrorHandler {
 
-    private final HttpServerResponse response;
+    private final RoutingContext routingContext;
 
-    HttpResponseErrorHandler(final HttpServerResponse response) {
-      this.response = response;
+    HttpResponseErrorHandler(final RoutingContext routingContext) {
+      this.routingContext = Objects.requireNonNull(routingContext);
     }
 
     @Override
     public void onMissingParam(final String paramName) {
-      handleError(response, 400, ErrorCodes.ERROR_CODE_MISSING_PARAM,
-          "No " + paramName + " in arguments");
+      routingContext
+          .fail(HttpStatus.SC_BAD_REQUEST, new KsqlApiException("No " + paramName + " in arguments",
+              ERROR_CODE_MISSING_PARAM));
     }
 
     @Override
     public void onExtraParam(final String paramName) {
-      handleError(response, 400, ErrorCodes.ERROR_CODE_UNKNOWN_PARAM,
-          "Unknown arg " + paramName);
+      routingContext
+          .fail(HttpStatus.SC_BAD_REQUEST, new KsqlApiException("Unknown arg " + paramName,
+              ErrorCodes.ERROR_CODE_UNKNOWN_PARAM));
```

Comment:
nit: inconsistent static imports. `ERROR_CODE_MISSING_PARAM` above is a static import but this error code (and the one below) aren't.